src/SerializeAST.cc

```diff
@@ -554,7 +582,7 @@ sos::Object WrapAction(const Action& action)
     return actionObject;
 }
 
-sos::Object WrapResource(const Resource& resource)
+sos::Object WrapResource(const Resource& resource, bool printElementValue = false)
```

Comment:
When using this function to print a resource in one of the resourceGroups, we should not print `"element": "resource"`, where as in the other case, we need to.